pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -8,6 +8,24 @@ mod v4;
 mod v5;
 mod v6;
 
+#[cfg(feature = "try-runtime")]
+pub(crate) fn pre_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
+	// potentially do some checks on the state before the migration
+	if version < 6 {
+		v6::pre_upgrade::<T>()?;
+	}
+	Ok(())
+}
+
+#[cfg(feature = "try-runtime")]
+pub(crate) fn post_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
```

Comment:
When is the storage version incremented? Doesn't post_upgrade means that the version has been incremented? Why is the condition `version < 6` instead of 7?

Reply:
Storage version is incremented during the migration. The version here is passed from `pre_upgrade` to `post_upgrade` (basically the bytes returned in `pre_upgrade` just get passed to `post_upgrade` to allow sharing state).

It is confusing though, it probably is more intuitive to read from storage - it just felt silly to have

```rust
let version = StorageVersion::get();

if version == 6 {
    // this is basically the only check in v6::post_upgrade right now
    assert_eq!(StorageVersion::get(), 6);
}
```

since this would break if we didn't correctly update the `StorageVersion`

Reply:
> since this would break if we didn't correctly update the StorageVersion

Ha! I am in favor then. We could potentially make a mistake so more breaking stuff in testing is better.